integration_tests/tests/integration/janus.rs

```diff
@@ -443,3 +458,60 @@ async fn janus_in_process_one_round_with_agg_param_time_interval() {
     )
     .await;
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn janus_in_process_histogram_dp_noise() {
+    static TEST_NAME: &str = "janus_in_process_histogram_dp_noise";
+    const HISTOGRAM_LENGTH: usize = 100;
+    const CHUNK_LENGTH: usize = 10;
+
+    install_test_trace_subscriber();
+    initialize_rustls();
+
+    let epsilon = Rational::from_unsigned(1u128, 10u128).unwrap();
+    let janus_pair = JanusInProcessPair::new(TaskBuilder::new(
+        QueryType::TimeInterval,
+        VdafInstance::Prio3Histogram {
+            length: HISTOGRAM_LENGTH,
+            chunk_length: CHUNK_LENGTH,
+            dp_strategy: vdaf_dp_strategies::Prio3Histogram::PureDpDiscreteLaplace(
+                PureDpDiscreteLaplace::from_budget(PureDpBudget::new(epsilon).unwrap()),
+            ),
+        },
+    ))
+    .await;
+    let vdaf = Prio3::new_histogram_multithreaded(2, HISTOGRAM_LENGTH, CHUNK_LENGTH).unwrap();
+
+    let total_measurements: usize = janus_pair
+        .task_parameters
+        .min_batch_size
+        .try_into()
+        .unwrap();
+    let measurements = iter::repeat(0).take(total_measurements).collect::<Vec<_>>();
+    let client_implementation = ClientBackend::InProcess
+        .build(
+            TEST_NAME,
+            &janus_pair.task_parameters,
+            (janus_pair.leader.port(), janus_pair.helper.port()),
+            vdaf.clone(),
+        )
+        .await
+        .unwrap();
+    let before_timestamp = submit_measurements_generic(&measurements, &client_implementation).await;
+    let (report_count, aggregate_result) = collect_aggregate_result_generic(
+        &janus_pair.task_parameters,
+        janus_pair.leader.port(),
+        vdaf,
+        before_timestamp,
+        &(),
+    )
+    .await;
+    assert_eq!(report_count, janus_pair.task_parameters.min_batch_size);
+
+    let mut un_noised_result = [0u128; HISTOGRAM_LENGTH];
+    un_noised_result[0] = report_count.into();
+    // Smoke test: Just confirm that some noise was added. Since epsilon is small, the noise will be
+    // large (drawn from Laplace_Z(20) + Laplace_Z(20)), and it is highly unlikely that all 100
```

Comment:
Naive: what is `_Z`? (you can point me to a paper :D).

Reply:
This is a lo-fi version of the notation $Lap_\mathbb{Z}(t)$ from section 4 of https://arxiv.org/pdf/2004.00010. (where blackboard Z represents the set of the integers)

Reply:
Also naive: there is no bound on the noise that could be applied to the result, yes? I'm mildly uncomfortable that the set of all valid results for this test is "all integers except for `n`", but seeing as the test fails if I turn off DP, I'm fine with it.

Reply:
Correct, the noise value could be any integer, thus we could be adding any field elements to the shares. (Though the probabilities become astronomically low once you get many multiples of the scale parameter out)

I also did a negative test where I cranked epsilon way up (to `u128::MAX`) and confirmed that the test failed then. Even though it was still adding noise, the noise was all zeros because of the absurd parameter choice.

We could improve this with some sort of statistical test that takes each dimension of the noise vector as one sample of the noise distribution, and then performs a statistical test to see how good of a fit it is for the desired distribution. (with a low enough false positive rate to not cause problems) I filed #3325 to track this as a follow-up. It'll be a bit tricky because the distribution of the sum of two numbers drawn from base distributions is the convolution of those distributions, plus I'll need to read up on statistical tests.